addons/imaginary.sh

```diff
@@ -0,0 +1,144 @@
+then
+    print_text_in_color "$IGreen" "imaginary seems to be working OK!"
+else
+    msg_box "Test failed, please report this to: $ISSUES"
+    exit
+fi
+
+# Install preview generator just in case
+install_and_enable_app previewgenerator
+
+# check if the previewgenerator is installed and enabled
+if is_app_enabled previewgenerator
+then
+    # enable previews
+    nextcloud_occ config:system:set enable_previews --value=true --type=boolean
+
+    # install needed dependency for movies
+    install_if_not ffmpeg
+else
+    exit
+fi
+
+# Set providers
+nextcloud_occ config:system:set enabledPreviewProviders 0 --value="OC\\Preview\\JPEG"
+nextcloud_occ config:system:set enabledPreviewProviders 1 --value="OC\\Preview\\PNG"
+nextcloud_occ config:system:set enabledPreviewProviders 2 --value="OC\\Preview\\WEBP"
+nextcloud_occ config:system:set enabledPreviewProviders 3 --value="OC\\Preview\\HEIF"
+nextcloud_occ config:system:set enabledPreviewProviders 4 --value="OC\\Preview\\TIFF"
+nextcloud_occ config:system:set enabledPreviewProviders 5 --value="OC\\Preview\\PDF"
+nextcloud_occ config:system:set enabledPreviewProviders 6 --value="OC\\Preview\\GIF"
+nextcloud_occ config:system:set enabledPreviewProviders 7 --value="OC\\Preview\\SVG"
+nextcloud_occ config:system:set enabledPreviewProviders 8 --value="OC\\Preview\\MP3"
+nextcloud_occ config:system:set enabledPreviewProviders 9 --value="OC\\Preview\\TXT"
+nextcloud_occ config:system:set enabledPreviewProviders 10 --value="OC\\Preview\\MarkDown"
+nextcloud_occ config:system:set enabledPreviewProviders 11 --value="OC\\Preview\\OpenDocument"
+nextcloud_occ config:system:set enabledPreviewProviders 12 --value="OC\\Preview\\Krita"
+nextcloud_occ config:system:set enabledPreviewProviders 13 --value="OC\\Preview\\BMP"
+nextcloud_occ config:system:set enabledPreviewProviders 14 --value="OC\\Preview\\Movie"
+nextcloud_occ config:system:set enabledPreviewProviders 15 --value="OC\\Preview\\Imaginary"
+nextcloud_occ config:system:set preview_imaginary_url --value="http://127.0.0.1:9000"
+
+# Set values
+nextcloud_occ config:app:set previewgenerator squareSizes --value="32 256"
+nextcloud_occ config:app:set previewgenerator widthSizes  --value="256 384"
+nextcloud_occ config:app:set previewgenerator heightSizes --value="256"
+nextcloud_occ config:system:set preview_max_x --value="2048"
+nextcloud_occ config:system:set preview_max_y --value="2048"
+nextcloud_occ config:system:set jpeg_quality --value="60"
+nextcloud_occ config:system:set preview_max_memory --value="128"
+nextcloud_occ config:app:set preview jpeg_quality --value="60"
+
+# Add logs
+touch "$VMLOGS"/previewgenerator.log
+chown www-data:www-data "$VMLOGS"/previewgenerator.log
+
+# Rebuild
+print_text_in_color "$ICyan" "Scanning Nextclouds appdata directory to rebuild all previews all previews. This will take a while..."
+nextcloud_occ files:scan-app-data -vvv
+nextcloud_occ preview:generate-all --verbose >> "$VMLOGS"/previewgenerator.log
```

Comment:
Not needed...

Reply:
Same with height size

Reply:
But I want to scan all the existing data as well... I took it from this post: https://okxo.de/speed-up-nextcloud-preview-generation-with-imaginary/

Reply:
Hm... Imaginary is not coupled with the previewgenerator...

Reply:
Ok, so a better way forward would be to uninstall previewgeneration and cleanup all generated previews, then the run a appfolder scan, and let imaginary handle the rest?